src/gui/vector/qgsfieldcalculator.h

```diff
@@ -43,7 +43,7 @@ class GUI_EXPORT QgsFieldCalculator: public QDialog, private Ui::QgsFieldCalcula
 {
     Q_OBJECT
   public:
-    QgsFieldCalculator( QgsVectorLayer *vl, QWidget *parent = nullptr );
+    QgsFieldCalculator( QgsVectorLayer *vl, QWidget *parent = nullptr, int fieldIndex = -1 );
```

Comment:
```suggestion
    /**
     * Constructor for QgsFieldCalculator, with the specified \a parent widget.
     * 
     * The target layer must be specified using the \a vl argument.
     * 
     * Since QGIS 3.36, the optional \a fieldIndex argument can be used to automatically
     * select the existing field with the specified index in the dialog. 
     */
    QgsFieldCalculator( QgsVectorLayer *vl, QWidget *parent = nullptr, int fieldIndex = -1 );
```

Reply:
(You'll need to re-run sipify after making this change)